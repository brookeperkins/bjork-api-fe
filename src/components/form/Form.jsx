import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ image, year, quote, onChange, onSubmit, doneDisplay }) => (
  <div key="form-div-key" className="form-div">

    <form onSubmit={onSubmit} className="create-outfit-form">
      <div className="form-header">
        <h1>Share Your Favorite Bjork Outfit</h1>
      </div>

      <div className="enter-image">
        <label>Image url:
          <input 
            type="text" 
            name="image" 
            placeholder="" 
            value={image} 
            onChange={onChange} />
        </label>
      </div>

      <div className="enter-year">
        <label>What year is it from?
          <input 
            type="number" 
            name="year" 
            placeholder="" 
            value={year} 
            onChange={onChange} />
        </label>
      </div>

      <div className="enter-quote">
        <label>Enter a quote:
          <input 
            type="text" 
            name="quote" 
            placeholder="" 
            value={quote} 
            onChange={onChange} />
        </label>
      </div>

      <button data-testid="form-button">Submit</button>
    </form>

    <div data-testid="done-display">
      <p>{doneDisplay}</p>
    </div>

  </div>
);

Form.propTypes = {
  image: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  quote: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  doneDisplay: PropTypes.func.isRequired
};

export default Form;
