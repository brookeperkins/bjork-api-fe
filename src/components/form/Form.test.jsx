import React from 'react';
import { render } from '@testing-library/react';
import Form from './Form';

describe('<Form />', () => {
  const number = 1994;

  it('renders the form', () => {
    const { asFragment } = render(<Form
      image="https://bjork.com/bjork.jpg"
      year={number}
      quote="Bjork quote here."
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
