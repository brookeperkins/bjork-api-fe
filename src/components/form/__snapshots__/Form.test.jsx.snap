// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the form 1`] = `
<DocumentFragment>
  <div
    class="form-div"
  >
    <form
      class="create-outfit-form"
    >
      <div
        class="form-header"
      >
        <h1>
          Share Your Favorite Bjork Outfit
        </h1>
      </div>
      <div
        class="enter-image"
      >
        <label>
          Image url:
          <input
            name="image"
            placeholder=""
            type="text"
            value="https://bjork.com/bjork.jpg"
          />
        </label>
      </div>
      <div
        class="enter-year"
      >
        <label>
          What year is it from?
          <input
            name="year"
            placeholder=""
            type="number"
            value="1994"
          />
        </label>
      </div>
      <div
        class="enter-quote"
      >
        <label>
          Enter a quote:
          <input
            name="quote"
            placeholder=""
            type="text"
            value="Bjork quote here."
          />
        </label>
      </div>
      <button
        data-testid="form-button"
      >
        Submit
      </button>
    </form>
    <div
      data-testid="done-display"
    >
      <p />
    </div>
  </div>
</DocumentFragment>
`;
